### PR TITLE
[rpc] add viewID and epoch to RPCMarshalBlock

### PR DESCRIPTION
**Pull request by ivorytowerdds**
want to get the number of block numbers in each epoch for non-beacon shards, so add epoch number to `hmy_getBlockByNumber`.

### CODE REVIEW

File: internal/hmyapi/apiv1/types.go [hunk 1]
```diff
@@ -353,6 +353,8 @@ func newRPCStakingTransaction(tx *types2.StakingTransaction, blockHash common.Ha
 // RPCBlock represents a block that will serialize to the RPC representation of a block
 type RPCBlock struct {
 	Number           *hexutil.Big     `json:"number"`
+	ViewID           *hexutil.Big     `json:"viewID"`
```
Comment (gupadhyaya) [inline internal/hmyapi/apiv1/types.go:356]: why do you need these here? It is not used right? only sufficient to add to RPCMarshalBlock?

Comment (ivorytowerdds) [reply internal/hmyapi/apiv1/types.go:356]: yes, it's not used in RPCMarshalBlock and other places. But seems like we record all the other fields in RPCMarshalBlock here, so I add them here too. Do we want to keep the type RPCBlock?

Comment (gupadhyaya) [reply internal/hmyapi/apiv1/types.go:356]: ok. later rpc revamp will take care of this then.

**Comment by Daniel-VDM** [comment]
Jenkins, test this please

**Comment by ivorytowerdds** [comment]
> Jenkins, test this please

how to restart the jenkins again?

**Comment by gupadhyaya** [comment]
@ivorytowerdds do you still need this PR?

**Pull request merged**